apis/install/v1alpha1/armadaserver_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ArmadaServerSpec defines the desired state of ArmadaServer
+type ArmadaServerSpec struct {
+	// Name specifies the base name for all Kubernetes Resources
+	Name string `json:"name"`
+	// Labels is the map of labels which wil be added to all objects
+	Labels map[string]string `json:"labels,omitempty"`
+	// Image is the configuration block for the image repository and tag
+	Image Image `json:"image"`
+	// ApplicationConfig is the internal ArmadaServer configuration which will be created as a Kubernetes Secret and mounted in the Kubernetes Deployment object
+	ApplicationConfig runtime.RawExtension `json:"applicationConfig"`
+	// PrometheusConfig is the configuration block for Prometheus monitoring
+	Prometheus *PrometheusConfig `json:"prometheus,omitempty"`
+	// Resources is the configuration block for setting Executor resource requirements
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	// Tolerations is the configuration block for specifying which taints the ArmadaServer pod can tolerate
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// TerminationGracePeriodSeconds specifies how many seconds should Kubernetes wait for the application to shut down gracefully before sending a KILL signal
+	TerminationGracePeriodSeconds int `json:"terminationGracePeriodSeconds,omitempty"`
```

Comment:
```suggestion
	TerminationGracePeriodSeconds *int64 `json:"terminationGracePeriodSeconds,omitempty"`
```

Following convetion in other APIs as of lastest main